plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -559,6 +565,26 @@ private HasMetadata convertKubernetesItemToOpenShift(HasMetadata item) {
         // lets check if there's an OpenShift resource of this name already from a dependency...
         HasMetadata dependencyResource = openshiftDependencyResources.convertKubernetesItemToOpenShift(item);
         if (dependencyResource != null) {
+            // this resource won't have been processed by the NamespaceEnricher yet so lets enrich it
+            try {
+                EnricherContext.Builder ctxBuilder = createEnricherContextBuilder();
+                EnricherContext enricherContext = ctxBuilder.build();
+
+                // TODO should we could replace this with a custom profile for namespace enricher?
+                NamespaceEnricher enricher = new NamespaceEnricher(enricherContext);
```

Comment:
'guess you already know what I would comment on this, so I'd better leave it alone. Doesn't make sense to repeat myself endlessly.

I know someone is supposed to eventually clean this up at some point in time. It can always be done later.

Reply:
@rhuss : Does this PR still makes sense in current context? If yes, we can do that required refactoring and cherry-pick this to master. WDYT?